sentry/src/routes/channel.rs

```diff
@@ -846,4 +857,86 @@ mod test {
             balances.spenders.get(&CREATOR),
         );
     }
+
+    #[tokio::test]
+    async fn get_channels_list() {
+        let mut app = setup_dummy_app().await;
+        app.config.channels_find_limit = 2;
+
+        let channel = Channel {
+            leader: IDS[&LEADER],
+            follower: IDS[&FOLLOWER],
+            guardian: *GUARDIAN,
+            token: "0x6B175474E89094C44Da98b954EedeAC495271d0F".parse()
+            .expect("Valid DAI token address"),
+            nonce: Nonce::from(987_654_321_u32),
+        };
+        insert_channel(&app.pool, channel).await.expect("should insert");
+        let channel_other_token = Channel {
+            leader: IDS[&LEADER],
+            follower: IDS[&FOLLOWER],
+            guardian: *GUARDIAN,
+            token: "0x73967c6a0904aa032c103b4104747e88c566b1a2".parse()
+            .expect("Valid DAI token address"),
+            nonce: Nonce::from(987_654_321_u32),
+        };
+        insert_channel(&app.pool, channel_other_token).await.expect("should insert");
+
+        let channel_other_leader = Channel {
+            leader: IDS[&LEADER_2],
+            follower: IDS[&FOLLOWER],
+            guardian: *GUARDIAN,
+            token: "0x6B175474E89094C44Da98b954EedeAC495271d0F".parse()
+            .expect("Valid DAI token address"),
+            nonce: Nonce::from(987_654_321_u32),
+        };
+        insert_channel(&app.pool, channel_other_leader).await.expect("should insert");
+
+        let build_request = |query: ChannelListQuery| {
+            let query = serde_urlencoded::to_string(query).expect("should parse query");
+            Request::builder()
+                .uri(format!("http://127.0.0.1/v5/channel/list?{}", query))
+                .extension(DUMMY_CAMPAIGN.channel.clone())
+                .body(Body::empty())
+                .expect("Should build Request")
+        };
+
+        // Test query page - page 0, page 1
+        let query = ChannelListQuery {
+            page: 0,
+            validator: None,
+        };
+        let res = channel_list(build_request(query), &app).await.expect("should get channels");
+        let channels_list = res_to_channel_list_response(res).await;
+        assert_eq!(channels_list.channels.len(), 2, "There should be 2 channels on the first page");
```

Comment:
We should be asserting the exact channels we expect instead of just `len()` for all tests.

They should be all returned with `ORDER BY created ASC` so assert the order of the channels